web/src/services/auth/index.tsx

```diff
@@ -18,3 +18,13 @@ export const AuthenticationRequiredPage: React.FC<{ children?: ReactNode }> = ({
     </>
   ) : null;
 };
+
+export const withAuthorisation = (): ((props: any) => React.FC<any>) => {
+  const authProvider = window.REEARTH_CONFIG?.authProvider;
+
+  if (authProvider === "cognito") {
+    return withAuthenticator as unknown as (props: any) => React.FC<any>;
+  }
+
+  return withAuthenticationRequired as unknown as (props: any) => React.FC<any>;
```

Comment:
Couple questions:
1. Do we really need to type like this? (#2 might solve this)
2. Can we encapsulate `const withAuthenticationFun = withAuthorisation();` here so that we just do `export default withAuthentication(ProjectPage);` inside each component?

Reply:
1. I had to, I tried to go with type like `((props: any) => JSX.Element)` which didn't worked it. Do you mean https://github.com/reearth/reearth-web/pull/2 ?
2. Something like this https://github.com/reearth/reearth/pull/514/commits/aa05523092fd7a75242d8fb8c8188ce311f03123 ?